ui/src/commonMain/kotlin/com/materiiapps/gloom/ui/screens/auth/LandingScreen.kt

```diff
@@ -19,56 +21,115 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import cafe.adriel.voyager.core.screen.Screen
 import cafe.adriel.voyager.koin.getScreenModel
+import cafe.adriel.voyager.navigator.LocalNavigator
+import cafe.adriel.voyager.navigator.currentOrThrow
 import com.materiiapps.gloom.Res
+import com.materiiapps.gloom.domain.manager.Account
+import com.materiiapps.gloom.ui.components.DividerWithLabel
+import com.materiiapps.gloom.ui.components.LoadingButton
 import com.materiiapps.gloom.ui.icons.GitHub
 import com.materiiapps.gloom.ui.icons.Social
+import com.materiiapps.gloom.ui.screens.root.RootScreen
 import com.materiiapps.gloom.ui.viewmodels.auth.LandingViewModel
+import com.materiiapps.gloom.ui.widgets.accounts.AccountItem
+import com.materiiapps.gloom.utils.LocalLinkHandler
 import dev.icerock.moko.resources.compose.painterResource
 import dev.icerock.moko.resources.compose.stringResource
 
-class LandingScreen : Screen {
+class LandingScreen(
+    private val showAccountCard: Boolean = true
+) : Screen {
 
     @Composable
     override fun Content() = Screen()
 
     @Composable
-    private fun Screen(
-        viewModel: LandingViewModel = getScreenModel()
-    ) {
+    private fun Screen() {
+        val linkHandler = LocalLinkHandler.current
+        val viewModel: LandingViewModel = getScreenModel()
+        val nav = LocalNavigator.currentOrThrow
+
         Surface {
             Column(
                 modifier = Modifier.fillMaxSize(),
                 verticalArrangement = Arrangement.Center,
                 horizontalAlignment = Alignment.CenterHorizontally
             ) {
                 Spacer(Modifier.weight(0.20f))
+
                 Image(
                     painter = painterResource(Res.images.gloom_logo),
                     contentDescription = null,
                     modifier = Modifier.size(300.dp)
                 )
+
                 Spacer(modifier = Modifier.weight(0.10f))
+
                 Text(
                     text = stringResource(Res.strings.login_welcome),
                     style = MaterialTheme.typography.headlineLarge.copy(
                         fontWeight = FontWeight.Thin
                     ),
                     modifier = Modifier.padding(bottom = 24.dp)
                 )
-                Button(onClick = { viewModel.signIn() }) {
-                    Row(
-                        horizontalArrangement = Arrangement.spacedBy(ButtonDefaults.IconSpacing),
-                        verticalAlignment = Alignment.CenterVertically
+
+                if (viewModel.authManager.accounts.size > 0 && showAccountCard) {
+                    val accounts = viewModel.authManager.accounts.values.toList()
```

Comment:
this isn't stable
you should `remember(accounts) { derived { accounts.values.toImmutableList() } }`